### ISSUE TITLE
[Feature Request] Warn about dangerous Phpunit assertions, e.g. assertEquals()

### ISSUE THREAD

**Issue by andreasschroth** [kalessil/phpinspectionsea#1097]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Feature Request                          |
| Plugin         | Php Inspections (EA Ultimate), 2.0.9 |
| Language level | PHP 7.2                        |

Current behaviour:
---

Here is some example code:

```php
<?php
public function testFoo(): void
{
    self::assertEquals(1024, true);
}
```

Phpunit marks the above test as success.

No warning is given for this piece of code from Php Inspections (EA Ultimate).

Suggested behaviour:
---

I know, this is an opinionated and subjective thing, but I'd love to see a warning here from Php Inspections (EA Ultimate). I actually don't see any good reason for using assertEquals ever. Comparing ValueObjects might be a legit use-case, but also here I guess it's better the objects themselves implement a proper `equals()` method and then you can write something like (`Email` being some example value object here, I guess I don't have to add pseudo code for that class =) ):

```php
<?php
public function testBar(): void
{
    $email = new Email('foo@example.com');
    self::assertTrue($email->equals(new Email('foo@example.com')));
}
```

At least me, I don't like risky assertions that do some weird type conversion in the background. As the first example shows, some weird assertions can be created with that. Obviously it should also warn for similar assertions, e.g. `assertNotEquals()`.

**Comment by kalessil** [comment]
This has been addressed with: 'Suggest to use type safe asserts' option, if i understanding the request correctly.
![image](https://user-images.githubusercontent.com/1577185/48330785-1b9e5900-e64e-11e8-80fe-af0f88ba519d.png)

**Comment by andreasschroth** [comment]
Oh, awesome. Didn't see it in the options, thanks!

**Comment by kalessil** [comment]
No problems =) I'm also open for improving the option wording - perhaps is so bad, that you missed it ;)

**Comment by andreasschroth** [comment]
I actually think the wording is fine, I honestly just didn't check the options of that inspection at all...

**Comment by kalessil** [comment]
Oh, no worries then =)
